sklearn/utils/estimator_checks.py

```diff
@@ -513,9 +522,14 @@ def _should_be_skipped_or_marked(estimator, check):
     return False, "placeholder reason that will never be used"
 
 
-def parametrize_with_checks(estimators):
+def parametrize_with_checks(estimators, legacy=True):
     """Pytest specific decorator for parametrizing estimator checks.
 
+    Checks are categorised into the following groups:
+
+        - API checks: a set of checks to ensure API compatibility with scikit-learn
```

Comment:
Currently, it's hard to tell what "API checks" means without looking at the code.

As we begin to categorize the checks, can we document the checks?

Reply:
Yes, my plan is to sync the tests with what "scikit-learn API" is in the developer's guide.

Reply:
Is it worth linking here then? 

https://scikit-learn.org/dev/developers/develop.html#apis-of-scikit-learn-objects

Reply:
Yes, I'm working on fixing that documentation in a separate PR.

Reply:
I would add the link in this PR even though this is not yet aligned programmatically.

```suggestion
        - API checks: a set of checks to ensure API compatibility with scikit-learn.
          Refer to https://scikit-learn.org/dev/developers/develop.html a requirement of
          scikit-learn estimators.
```